crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -560,12 +578,21 @@ async fn execute_ics20_transfer_bridge_lock<S: StateWriteExt>(
         return Ok(());
     }
 
+    // assert memo is valid
+    let deposit_memo: Ics20TransferDepositMemo =
+        serde_json::from_str(&memo).context("failed to parse memo as Ics20TransferDepositMemo")?;
+
     ensure!(
-        !destination_address.is_empty(),
+        !deposit_memo.rollup_address.is_empty(),
```

Comment:
Does this really need extra checking? We can't really know if the destination address is valid or not: what if I put `"."` or `"_"` or `"\n"` in there?

Reply:
it must not be empty, otherwise there is no chance it's a valid rollup address, we can't check for every case obviously but nonempty is pretty simple